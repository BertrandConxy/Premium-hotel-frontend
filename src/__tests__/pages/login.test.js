import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import LogIn from "../../pages/LogIn";
import store from "../../redux/configureStore";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";

const LogInProvider = () => (
  <Provider store={store}>
    <Router>
      <LogIn />
    </Router>
  </Provider>
);

describe("render component", () => {
  it("renders the login form", () => {
    render(<LogInProvider />);
    const LogIn = screen.getByText("Log In");
    const SignUp = screen.getByText("Sign Up");
    expect(LogIn).toBeInTheDocument();
    expect(SignUp).toBeInTheDocument();
  });
});

describe("Snapshot test", () => {
  it("looks the same as the snapshot", () => {
    const component = renderer.create(<LogInProvider />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
