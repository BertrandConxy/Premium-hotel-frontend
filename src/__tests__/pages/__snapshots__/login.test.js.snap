// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test looks the same as the snapshot 1`] = `
<div
  className="flex h-screen bg-white"
>
  <div
    className="w-full max-w-sm m-auto bg-gray-200 rounded p-5"
  >
    <header>
      <img
        alt="logo"
        className="w-20 mx-auto mb-4"
        src="https://img.icons8.com/fluency-systems-regular/96/99c22a/5-star-hotel.png"
      />
    </header>
    <form
      className="col mt-12"
      onSubmit={[Function]}
    >
      <input
        className="w-full p-2 mb-6 border-b-2 bg-green outline-none focus:bg-gray-300"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email"
      />
      <input
        className="w-full p-2 mb-6 border-b-2 bg-green outline-none focus:bg-gray-300"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <input
        className="w-full bg-green-button text-white font-bold py-2 px-4 mb-6 rounded"
        type="submit"
        value="Log In"
      />
    </form>
    <small
      className="flex justify-center items-center"
    >
      Don't have an account?
      <a
        className="cursor-pointer bg-green px-2 text-sm"
        href="/users/signup"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </small>
  </div>
</div>
`;
